falcon/response.py

```diff
@@ -136,6 +145,82 @@ def set_stream(self, stream, stream_len):
         self.stream = stream
         self.stream_len = stream_len
 
+    def set_cookie(self, name, value, expires=None, max_age=None,
+                   domain=None, path=None, secure=True, httponly=True):
+        """Set a response cookie.
+
+        Args:
+            name (str):
+                Cookie name
+            value (str):
+                Cookie value
+            expires (datetime): Defines when the cookie should expire.
+                the default is to expire when the browser is closed.
+            max_age (int): The Max-Age attribute defines the lifetime of the
+                cookie, in seconds.  The delta-seconds value is a decimal non-
+                negative integer.  After delta-seconds seconds elapse,
+                the client should discard the cookie.
+            domain (str): The Domain attribute specifies the domain for
+                which the cookie is valid.
+                An explicitly specified domain must always start with a dot.
+                A value of 0 means the cookie should be discarded immediately.
+            path (str): The Path attribute specifies the subset of URLs to
+                which this cookie applies.
+            secure (bool) (default: True): The Secure attribute directs the
```

Comment:
@tbug OK, I think it makes sense to simply refuse all non-HTTPS requests given a flag is enabled (#460). The developer is responsible for enabling that flag in production deployments. This acts as a last-ditch check in case HTTP is accidentally allowed throw at the LB/server layer.

Cookies won't try to do any additional checking. However, I would like to see some warning in the docstring for the `secure` arg that the developer should be sure to enforce HTTPS in production deployments when using secure cookies.